crates/forge-runner/src/coverage_api.rs

```diff
@@ -49,3 +54,50 @@ pub fn run_coverage(saved_trace_data_paths: &[PathBuf]) -> Result<()> {
 
     Ok(())
 }
+
+pub fn can_coverage_be_generated(scarb_metadata: &Metadata) -> Result<()> {
+    let manifest = fs::read_to_string(&scarb_metadata.runtime_manifest)?.parse::<DocumentMut>()?;
+
+    let has_need_entries = manifest
+        .get("profile")
+        .and_then(|profile| profile.get(&scarb_metadata.current_profile))
+        .and_then(|profile| profile.get("cairo"))
+        .and_then(|cairo| cairo.as_table())
+        .is_some_and(|profile_cairo| {
+            [
+                ("unstable-add-statements-functions-debug-info", "true"),
+                ("unstable-add-statements-code-locations-debug-info", "true"),
+                ("inlining-strategy", "\"avoid\""),
+            ]
+            .iter()
+            .all(|(key, value)| contains_entry_with_value(profile_cairo, key, value))
+        });
+
+    ensure!(
+        has_need_entries,
+        formatdoc! {
+            "Scarb.toml must have the following Cairo compiler configuration to run coverage:
+
+            [profile.{profile}.cairo]
+            unstable-add-statements-functions-debug-info = true
+            unstable-add-statements-code-locations-debug-info = true
+            inlining-strategy = \"avoid\"
+            ... other entries ...
```

Comment:
```suggestion
            "Scarb.toml must contain the following Cairo compiler configuration to run coverage:

            [profile.{profile}.cairo]
            unstable-add-statements-functions-debug-info = true
            unstable-add-statements-code-locations-debug-info = true
            inlining-strategy = \"avoid\"
```

Reply:
I think I would leave that as it is, since people might think that configuration  should look exactly like this, which is not true.